src/units/attack_type.hpp

```diff
@@ -87,8 +87,18 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	void modified_attacks(unsigned & min_attacks,
 	                      unsigned & max_attacks) const;
 
+	/**
+	 * Select best damage type based on frequency count for replacement_type and based on highest damage for alternative_type.
+	 *
+	 * @param abil_list list of [damage_type] to check.
+	 * @param key_name name of attribute checked 'alternative_type' or 'replacement_type'.
+	 * @param resistance_abilities list of "resistance" abilities to check for each type of damage checked.
+	 */
+	std::optional<std::string> select_damage_type(const unit_ability_list& abil_list, const std::string& key_name, unit_ability_list resistance_abilities) const;
```

Comment:
I think `abil_list` should be called `damage_type_abilities` similar to `resistance_abilities`. There are a couple more places `abil_list` is used where it should be renamed as well.

Reply:
renamed to damage_type_list

Reply:
Is there a reason you want to name both lists differently? My point was to name the variable for damage_type and resistance the same way.

Reply:
changed to resistance_list